morpheus-graphql-client/src/Data/Morpheus/Client/Declare/Type.hs

```diff
@@ -35,9 +35,14 @@ import Data.Morpheus.Types.Internal.AST
 import Language.Haskell.TH
 import Relude hiding (Type)
 
-typeDeclarations :: TypeKind -> [ClientTypeDefinition -> Q Dec]
-typeDeclarations KindScalar = []
-typeDeclarations _ = [pure . declareType]
+typeDeclarations :: TypeKind -> ClientTypeDefinition -> Q [Dec]
+typeDeclarations KindScalar _ = pure []
+typeDeclarations _ c@ClientTypeDefinition{clientTypeName = TypeNameTH{namespace, typename}} = do
+    let name = mkConName namespace typename
+    m <- lookupTypeName (show name)
```

Comment:
can we generalise lookup process?

Reply:
you mean extracting this function to another module and accept the name as an argument?

Reply:
i mean that we lookup type 2 times and both case little different way. we could write some util function. `isTypeDeclared` which would accept `clientTypeName` and use them in both places `Declare/Aeson.hs` and here